### PR TITLE
Implement the experimental pointer lock API

### PR DESCRIPTION
**Pull request by ryanisaacg**

### CODE REVIEW

File: src/webapi/events/pointer.rs [hunk 1]
```diff
@@ -358,6 +358,40 @@ impl ConcreteEvent for LostPointerCaptureEvent {
...
+// http://www.w3.org/TR/pointerlock/#pointerlockchange-and-pointerlockerror-events
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "Event")] // TODO: Better type check.
+#[reference(subclass_of(Event))]
+pub struct PointerLockChangeEvent( Reference );
+
+impl IEvent for PointerLockChangeEvent {}
+impl IUiEvent for PointerLockChangeEvent {}
+impl IMouseEvent for PointerLockChangeEvent {}
+impl IPointerEvent for PointerLockChangeEvent {}
```
Comment (koute) [inline src/webapi/events/pointer.rs]: Hmm... I don't think these events are supposed to derive from these? (The specs are somewhat unclear regarding that.)

Taking a peek a Firefox sources [it looks like it should just be an `Event`](https://github.com/mozilla/spiderweb/blob/2eca983d249b0c7573534d59a008747073aa5a33/dom/events/EventNameList.h#L345)? Can you double check?

Comment (ryanisaacg) [reply src/webapi/events/pointer.rs]: ...Huh. That's what I get for not checking my assumptions about web api's I guess. Yeah those derivations shouldn't be there.

**Comment by koute** [comment]
Also it'd be nice if you could add a few really basic tests just as we have for other events. (:

**Comment by ryanisaacg** [comment]
Sure, no problem.

**Comment by koute** [comment]
You also should export those events in `lib.rs` or else they'll be useless. (:

**Comment by ryanisaacg** [comment]
Whoops! My bad.

**Comment by koute** [comment]
LGTM; thanks!

**Pull request merged**